### PR TITLE
Export keeps feature

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster @andrewconner

### CODE REVIEW

**Comment by camhashemi** [comment]
@ryanpbrewster

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala [hunk 1]
```diff
@@ -725,31 +740,18 @@ class KeepCommanderImpl @Inject() (
...
+  private val librariesByUserId: Cache[Id[User], (Library, Library)] = CacheBuilder.newBuilder().concurrencyLevel(4).initialCapacity(128).maximumSize(128).expireAfterWrite(30, TimeUnit.SECONDS).build()
+  private def getLibFromPrivacy(isPrivate: Boolean, userId: Id[User])(implicit session: RWSession) = {
+    val (main, secret) = librariesByUserId.get(userId, new Callable[(Library, Library)] {
+      def call() = libraryInfoCommander.getMainAndSecretLibrariesForUser(userId)
+    })
+    if (isPrivate) {
+      secret
+    } else {
+      main
+    }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala]: Why was this added here? Is it needed?

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala [hunk 2]
```diff
@@ -117,6 +118,9 @@ class KeepCommanderImpl @Inject() (
     airbrake: AirbrakeNotifier,
     normalizedURIInterner: NormalizedURIInterner,
     clock: Clock,
+    libraryCommander: LibraryCommander,
+    libraryAccessCommander: LibraryAccessCommander,
+    libraryInfoCommander: LibraryInfoCommander,
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala]: It feels weird that we would need these in `KeepCommander`. Any idea why?

**Comment by ryanpbrewster** [comment]
I think your stuff lgtm. Some of my stuff seems a little weird. I tagged the things I think merit a second look.

Merge when ready.

**Comment by ryanpbrewster** [comment]
Just checked my branch. I'm pretty sure that some of the things I tagged are byproducts of the `git merge` of my branch into yours. They should be safe to delete.

**Comment by camhashemi** [comment]
Thanks @ryanpbrewster

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala (hunk 2 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala]: `.tagKeeps` checks `libraryAccessCommander` to know if a user can do that action

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala]: deleted some unused code with `libraryInfoCommander` and unused `libraryCommander`

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala]: removed... the merge was very messy
